stdio/src/lib.rs

```diff
@@ -51,27 +49,44 @@ where
 	/// The server reads from STDIN line-by-line, one request is taken
 	/// per line and each response is written to STDOUT on a new line.
 	pub fn build(&self) {
-		let stdin = tokio_stdin_stdout::stdin(0);
-		let stdout = tokio_stdin_stdout::stdout(0).make_sendable();
+		let stdin = tokio::io::stdin();
+		let mut stdout = tokio::io::stdout();
 
-		let framed_stdin = FramedRead::new(stdin, LinesCodec::new());
-		let framed_stdout = FramedWrite::new(stdout, LinesCodec::new());
+		let mut framed_stdin = FramedRead::new(stdin, LinesCodec::new());
 
 		let handler = self.handler.clone();
-		let future = framed_stdin
-			.and_then(move |line| Self::process(&handler, line).map_err(|_| unreachable!()))
-			.forward(framed_stdout)
-			.map(|_| ())
-			.map_err(|e| panic!("{:?}", e));
-
-		tokio::run(future);
+		use futures::StreamExt;
+		let future = async {
+			while let Some(request) = framed_stdin.next().await {
+				match request {
+					Ok(line) => {
+						let res = Self::process(&handler, line).await;
+						let mut sanitized = res.replace('\n', "");
+						sanitized.push('\n');
+						use tokio::io::AsyncWriteExt;
+						if let Err(e) = stdout.write_all(sanitized.as_bytes()).await {
+							log::warn!("Error writing response: {:?}", e);
+						}
+					}
+					Err(e) => {
+						log::warn!("Error reading line: {:?}", e);
+					}
+				}
+			}
+		};
+		let mut rt = tokio::runtime::Builder::new()
```

Comment:
Instead of spawning the runtime, maybe we could just return a future here?

Reply:
Sure; I just tried to adapt the existing code to work as before

Reply:
Done in a883edcdde94764b0efb19d579982f60ed8a029a